core/CMakeLists.txt

```diff
@@ -157,16 +157,10 @@ if(XCODE OR VS)
     ax_mark_code_files("${_AX_CORE_LIB}")
 endif()
 
+message("CMake ${_AX_CORE_LIB} target_precompile_headers")
+target_precompile_headers(${_AX_CORE_LIB} PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:cocos2d.h>")
+
 if(WINDOWS)
-    # precompiled header. Compilation time speedup ~4x, but don't know how Ninja to support PCH yet and needs help from someone.
-    if(NOT CMAKE_GENERATOR STREQUAL "Ninja")
```

Comment:
Ninja generator was borken?

Reply:
```
if(NOT CMAKE_GENERATOR STREQUAL "Ninja")
```
code was added by me in cocos2d-x  3.
I tried add support for Windows and Ninja. Currently I believe that for windows we should support only msvc compiler. 
Not many people use Ninja on windows for axmol/cocos2d-x - many things breaks...

Reply:
Can we merge it?
It speedup other platforms builds by ~ 30%?

Reply:
lgtm except Ninja generator was broken